webapp/lib/controller.dart

```diff
@@ -157,25 +197,65 @@ void loadDashboardView() {
 }
 
 void loadPackageConfigurationView() {
-  var packages = model.packageConfigurationData.keys.toList();
+  _toggleNavActions(NavAction.ALLPROJECTS, false);
+  _toggleNavActions(NavAction.DASHBOARD, true);
+  var packages = new List<Map<String, String>>.from(model.activePackages.map((package) => new Map<String, String>.from({'id': package['id'], 'name': package['name']})));
   selectedPackage = router.routeParams['package'];
-  var configuratorView = new view.PackageConfiguratorView(packages, model.packageConfigurationData[selectedPackage]);
+  var configuratorView = new view.PackageConfiguratorView(packages, _findActivePackageConfigurationById(selectedPackage));
   view.contentView.renderView(configuratorView);
 }
 
 loadProjectConfigurationView() {
+  _toggleNavActions(NavAction.ALLPROJECTS, false);
+  _toggleNavActions(NavAction.DASHBOARD, true);
   view.navView.projectTitle = project?.projectName;
   view.navView.projectOrganizations = [''];
   view.contentView.renderView(new view.ProjectConfigurationView(model.projectConfigurationFormData, model.additionalProjectConfigurationLanguages));
 }
 
 void addPackage(String packageName) {
-  model.packageConfigurationData[packageName] = new model.Configuration()
-    ..availableTags = model.tags;
-  model.activePackages.add({'name': packageName, 'conversationsLink': '#/conversations', 'configurationLink': '#/package-configuration?package=$packageName',  'chartData': ''});
+  var id  = 'package-${new Uuid().v4().split('-')[0]}';
+  model.activePackages.add(
+    {
+      'id': id,
+      'name': packageName,
+      'conversationsLink': '#/conversations',
+      'configurationLink': '#/package-configuration?package=$id',
+      'chartData': '',
+      'configurationData': new model.Configuration().availableTags = model.tags
+    }
+  );
   model.availablePackages.removeWhere((package) => package['name'] == packageName);
 }
 
+void duplicatePackage(String packageId, String packageName) {
+  var originalPackageConfiguration = _findActivePackageConfigurationById(packageId);
+  var newId = 'package-${new Uuid().v4().split('-')[0]}';
+  model.activePackages.add(
+    {
+      'id': newId,
+      'name': '$packageName [COPY]',
+      'conversationsLink': '#/conversations',
+      'configurationLink': '#/package-configuration?package=$newId',
+      'chartData': '',
+      'configurationData': new model.Configuration()
+        ..availableTags = originalPackageConfiguration.availableTags
+        ..hasAllTags = originalPackageConfiguration.hasAllTags
+        ..containsLastInTurnTags = originalPackageConfiguration.containsLastInTurnTags
+        ..hasNoneTags = originalPackageConfiguration.hasNoneTags
+        ..suggestedReplies = originalPackageConfiguration.suggestedReplies
+        ..addsTags = originalPackageConfiguration.addsTags
```

Comment:
most of these maps/lists should be copied, otherwise if you change one of them, the same change will appear in other packages!

Reply:
done cloning properly